### ISSUE TITLE
Unable to profile ruby code, potentially ruby version related

### ISSUE THREAD

**Issue by looneym** [Granulate/gprofiler#444]
Hi there.

I'm attempting to profile a ruby application using gprofiler but am unable to get ruby-level profiling information. Gprofiler appears to be profiling it as a c application instead. The profiles all look like this:

![image](https://user-images.githubusercontent.com/12705417/186204746-d89e1493-b1f5-464c-8a9d-11e7f0609303.png)

Some of the frames in the profile e.g `rb_class_s_new` correspond to the actual c implementation of ruby itself https://apidock.com/ruby/Class/new

The application is a Rails app run by the passenger webserver. On investigating the logs I see lines indicating that rbspy is working:

```
[15:14:10] Profiling process 667611 with rbspy (cmdline=Passenger AppPreloader: /apps/intercom/current (forking...)                                           )
...
[15:14:11] Finished profiling process 667611 with rbspy
```

But also quite a lot of this:

```
[15:17:06] RbSpyProfiler: failed to profile process 3256 (flying-passenge)
Traceback (most recent call last):
  File "/app/gprofiler/profilers/profiler_base.py", line 119, in _wait_for_profiles
    result = future.result()
  File "/usr/lib/python3.8/concurrent/futures/_base.py", line 437, in result
    return self.__get_result()
  File "/usr/lib/python3.8/concurrent/futures/_base.py", line 389, in __get_result
    raise self._exception
  File "/usr/lib/python3.8/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/app/gprofiler/profilers/ruby.py", line 115, in _profile_process
    run_process(
  File "/app/gprofiler/utils/__init__.py", line 265, in run_process
    raise CalledProcessError(retcode, process.args, output=stdout, stderr=stderr)
gprofiler.exceptions.CalledProcessError: Command '['/app/gprofiler/resources/ruby/rbspy', 'record', '--silent', '-r', '11', '-d', '60', '--nonblocking', '--on-cpu', '--format=collapsed', '--file', '/tmp/gprofiler_tmp/tmpx889co01/rbspy.EXYENVfrgNpMUiPy.3256.col', '--raw-file', '/dev/null', '-p', '3256']' returned non-zero exit status 1.
stdout: b''
stderr: b'Press Ctrl+C to stop\nWarning: no profile samples were collected\nWrote raw data to /dev/null\nWriting formatted output to /tmp/gprofiler_tmp/tmpx889co01/rbspy.EXYENVfrgNpMUiPy.3256.col\nSomething went wrong while rbspy was sampling the process. Here\'s what we know:\n- get ruby VM state\n- Couldn\'t get ruby process state. Please open a GitHub issue for this and include the following information:\nversion: "2.4.4"\ncurrent thread address: Err(\n    "Current thread address not found in process 3256",\n)\nVM address: Err(\n    "Could not find address for symbol ruby_current_vm",\n)\nglobal symbols address: Err(\n    "Could not find address for symbol global_symbols",\n)\n\n'
```

Ruby 2.4.4 is the system ruby installed on these hosts. The application is running on it's own ruby version of 2.7.5 (packaged in a tarball, running directly on the host). It seems like we might need to point gprofiler and rbspy at the correct ruby version/path but I've dug around in the options and can't see a way to do that. 

I have tried running gprofiler in the docker container you supply, as well as directly on the host using your "command line" install but I see the exact same behaviour in both cases. 

Output of `uname -a`
```
Linux <HOSTNAME REDACTED> 5.10.130-118.517.amzn2.aarch64 #1 SMP Wed Jul 13 16:51:59 UTC 2022 aarch64 aarch64 aarch64 GNU/Linux
```

**Comment by looneym** [comment]
I've switched to using `rbspy` directly using the latest version from their github (currently 0.12.1) which is working perfectly with no configuration.

**Comment by Jongy** [comment]
Hi @looneym !

The relevant error is `Something went wrong while rbspy was sampling the process. Here\'s what we know:\n- get ruby VM state\n- Couldn\'t get ruby process state. Please open a GitHub issue for this and include the following information:\nversion: "2.4.4"\ncurrent thread address: Err(\n    "Current thread address not found in process 3256",\n)\nVM address: Err(\n    "Could not find address for symbol ruby_current_vm",\n)\nglobal symbols address: Err(\n    "Could not find address for symbol global_symbols",\n)\n\n` indeed, and it teaches that rbspy was unable to handle the Ruby version.

rbspy doesn't care about different Ruby installations - per Ruby process it will find the ruby executable that powers it, and will get the symbols & offsets from that file. I'd think in your case (and your second comment proves it) that the problem is we're using a too old version of rbspy, and thus it fails on your "too new" 2.7.5 (indeed if I check rbspy 0.8.1 which is what we're using, it doesn't include support for Ruby 2.7.5).

So that's totally on us and the current blocker on our side for upgrading is to get https://github.com/rbspy/rbspy/pull/327 merged because gProfiler's usage of rbspy depends on this logic (we currently have it merged in our fork but we prefer to mainline). Once we finish with that, we'll be able to upgrade our rbspy and you'll have the profiler working fine :)

I'll be sure to update here when that happens, @looneym .

**Comment by nenaddzambasevic** [comment]
@Jongy any updates? I see that there is no updates on https://github.com/rbspy/rbspy/pull/327 for more than 2 months.

**Comment by Jongy** [comment]
@nenaddzambasevic indeed, there was a breaking change as noted [here](https://github.com/rbspy/rbspy/pull/327#issuecomment-1253575285). We haven't finished that merge yet.

Meanwhile we are based on rbspy 0.12.1 and I'll try to get it merged for the next release (see #396 ). Will that help in your use case? It will work for 2.7.5 as reported in this issue, but not for versions that rbspy added support for since 0.12.1.

**Comment by Jongy** [comment]
Hey @nenaddzambasevic and @looneym , gProfiler 1.15.0 was released yesterday and it uses rbspy 0.12.1 which can profile Ruby 2.7.5.

**Comment by nenaddzambasevic** [comment]
Hi @Jongy, we are using ruby 3.0.4 now but it looks it's supported since rbspy 0.12.0 so we will give it a go again to test out gProfiler.
